docs/typescript.md

```diff
@@ -114,9 +67,13 @@ module.exports = {
 
 5. Run `yarn tsc` to type-check your new TypeScript files.
 
+## Using JavaScript Instead of TypeScript
+
+React Native defaults new applications to TypeScript, but JavaScript may still be used by default. Files with a `.jsx` extension are treated as JavaScript instead of TypeScript, and will not be typechecked. JavaScript modules may still be imported by TypeScript modules, along with the reverse.
```

Comment:
```suggestion
React Native defaults new applications to TypeScript, but JavaScript may still be used. Files with a `.jsx` extension are treated as JavaScript instead of TypeScript, and will not be typechecked. JavaScript modules may still be imported by TypeScript modules, along with the reverse.
```